BB-4552: Add support for fixed rate for all products

alfredchavez opened the pull request:
This PR adds support for a fixed tax rate setting in ecommerce through a new django settings (
TAX_RATE), which default value is 0.0, preserving the previous platform behaviour.

Testing instructions:
- Check that the added tests make sense
- Change the setting to a decimal value(like "0.3") and verify that the total value displayed on basket and invoice includes the tax rate, could be tested with the verified certificate purchase

giovannicimolin reviewed (commented):
@alfredchavez Nice! It's looking good and working as expected, but I think we need a bit more depth in the tests.

Can you add a test that uses a basket and check if the tax strategy is properly set? 
Doesn't need anything too complex though, 1 extra test should do it (maybe an e2e one?).

Commit:
feat: Add support for fixed tax rate in ecommerce

This commit adds support for a fixed tax rate
for all e-commerce transactions. The rate can be adjusted
through `settings.TAX_RATE` and the default value of this
setting preserves the previous behaviour(0.0 tax rate)

alfredchavez commented:
@giovannicimolin applied new changes, let me know if something else is needed

alfredchavez commented on ecommerce/extensions/partner/strategy.py:
```diff
@@ -33,8 +35,18 @@ def availability_policy(self, product, stockrecord):
+    """
+
+    @property
+    def rate(self):
```
Here I changed it to @property instead of the class variable to make easier testing with override_settings, also I tested this in the UI and it is working fine

giovannicimolin reviewed (approved):
:+1:

- [x] I tested this:
1. Set a tax rate in Django settings.
2. Purchased a course and checked that the basket value includes tax :heavy_check_mark: 
3. Checked that the invoices included price with tax  :heavy_check_mark: 
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~ NA
- [ ] Includes documentation: follow up task.

@alfredchavez Approved conditionally on squashing both commits and following OEP-51.
Something like this:
```
feat: Add support for fixed tax rate

This commit adds support for a fixed tax rate
for all e-commerce transactions. The rate can be adjusted
through `settings.TAX_RATE` and the default value
preserves the previous behavior.
```
